www/js/lib/zimArchive.js

```diff
@@ -223,9 +223,10 @@ define(['zimfile', 'zimDirEntry', 'util', 'utf8'],
         util.binarySearch(0, this._file.articleCount, function(i) {
             return that._file.dirEntryByTitleIndex(i).then(function(dirEntry) {
                 if (search.status === 'cancelled') return 0;
-                if (dirEntry.namespace < 'A') return 1;
-                if (dirEntry.namespace > 'A') return -1;
-                // We should now be in namespace A
+                var ns = dirEntry.namespace;
+                if (ns < 'A') return 1;
+                if (ns === 'B' || ns > 'C') return -1;
```

Comment:
I compared the two specs. The old-format spec for namespaces is here: https://wiki.openzim.org/wiki/ZIM_file_format_old_namespace#Namespaces , and the new one is here: https://wiki.openzim.org/wiki/ZIM_file_format#Namespaces .

The new format can only have namespaces C,M,W,X. The old format has namespaces -,A,B,I,J,M,U,V,W,X. I guess C was chosen precisely because there is no ambiguity with earlier content namespaces.

I'll check libzim if I can find the corresponding code (I'm struggling to find the random function anywhere in libzim!). But this is the most economical logical test that works universally for both ZIM types. If a future spec is introduced, we'll need to change this code anyway.

Reply:
PS I will certainly add a comment.

Reply:
@mossroy My lack of cpp knowledge is hampering me. I found a long explanation here:

https://github.com/openzim/libzim/blob/master/src/archive.cpp#L217 

And there is a test function `if (m_impl->hasNewNamespaceScheme())` being called here (and in lots of places in this script):

https://github.com/openzim/libzim/blob/master/src/archive.cpp#L217

But I can't find any function actually corresponding to that.

And `search.cpp` seems to be full of Xapian-specific code.

Reply:
They seem to read the "minor version" in the header of the ZIM file to know if the ZIM file is using the new namespace sheme or not. See https://github.com/openzim/libzim/blob/93464a63019619fa9c9ef71401d2491df75bd189/src/fileimpl.cpp#L200 and https://openzim.org/wiki/ZIM_file_format#Header

It looks like a good and reliable idea. It should be easy to add that in our zimfile.js, so that we can use this information to know if we have to use the old way or new way to parse the ZIM file

Reply:
OK I think I can implement that. Should be much easier than adding the MIME type table was.

Reply:
Done.